simfit_data_fullAngularMass_Swave.cc

```diff
@@ -164,19 +164,68 @@ void simfit_data_fullAngularMass_SwaveBin(int q2Bin, int parity, bool multiSampl
   // Random generators
   RooRandom::randomGenerator()->SetSeed(1);
 
+  //
+  // retrieve Z4430 model workspace from file
+  // only a single year sample of Z is available, therefore the same parametrisation is used for the 3 years of data taking
+  //
+  RooAbsPdf* Z4430_ang_pdf = 0;
+  RooArgSet* Z4430_ang_params = 0;
+  RooAbsPdf* Z4430_mass_pdf = 0;
+  RooArgSet* Z4430_mass_params = 0;
+  if (q2Bin==6 && fitOption>0){
+    string filename_Z4430 = "/eos/user/a/aboletti/BdToKstarMuMu/fileIndex/Zmodel/HistZ4430.root";
+    if (!(retrieveWorkspace(filename_Z4430, wsp_Z4430, "wZ4430"))) return;
+
+    // retrieve angular component of the Z pdf
+    Z4430_ang_pdf = (RooAbsPdf*) wsp_Z4430[0]->pdf("Z4430_ang_pdf");
+    if(!Z4430_ang_pdf){
+      std::cout << "Z4430_ang_pdf not found!!!\n" << std::endl;
+      wsp_Z4430[0]->allPdfs();
+      exit(1);
+    }
+    Z4430_ang_params = (RooArgSet*) Z4430_ang_pdf->getParameters(RooArgSet(*ctK,*ctL,*phi));
+    auto iter_z = Z4430_ang_params->createIterator();
+    RooRealVar* ivar_z = (RooRealVar*)iter_z->Next();
+    int iii=1;
+    while (ivar_z) {
+      ivar_z->setConstant(true);
+      cout<<Form("ang par. %d) %s = %f\n",iii,ivar_z->GetName(),ivar_z->getVal())<<endl;
+      ivar_z = (RooRealVar*) iter_z->Next();
+      iii++;
+    }
+    // retrieve mass component of the Z pdf
+    Z4430_mass_pdf = (RooAbsPdf*) wsp_Z4430[0]->pdf("Z4430_mass_pdf");
+    if (!Z4430_mass_pdf){
+      std::cout << "Z4430_mass_pdf not found!!!\n" << std::endl;
+      wsp_Z4430[0]->allPdfs();
+      exit(1);
+    }
+    Z4430_mass_params = (RooArgSet*) Z4430_mass_pdf->getParameters(RooArgSet(*ctK,*ctL,*phi));
```

Comment:
I guess here it is meant to be:

```suggestion
    Z4430_mass_params = (RooArgSet*) Z4430_mass_pdf->getParameters(RooArgSet(*mass));
```

Reply:
Yes, you suggestion sound correct and this could be and cut&paste error. Nevertheless it works...

Reply:
I confirm that this issue has not impact. I approve the suggestion

Reply:
Yes, I imagined that there is no real impact (the only side effect is that we run setConstant(true) on the variable mass, but I assume that it does not impact the fit). But let's include this fix for completeness.